packages/block-library/src/social-link/index.php

```diff
@@ -13,16 +13,17 @@
  * @return string Rendered HTML of the referenced block.
  */
 function render_core_social_link( $attributes ) {
-	$site = ( isset( $attributes['site'] ) ) ? $attributes['site'] : 'Icon';
-	$url  = ( isset( $attributes['url'] ) ) ? $attributes['url'] : false;
+	$site  = ( isset( $attributes['site'] ) ) ? $attributes['site'] : 'Icon';
+	$url   = ( isset( $attributes['url'] ) ) ? $attributes['url'] : false;
+	$label = ( isset( $attributes['label'] ) ) ? $attributes['label'] : __( 'Link to ' ) . core_social_link_get_name( $site );
```

Comment:
I believe that this should be using a placeholder:

https://codex.wordpress.org/I18n_for_WordPress_Developers#Placeholders

Otherwise, translators will only be provided the string "Link to " without contextual clues to know that it is being used in a concatenation. Furthermore, some translations may require that the position of the placeholder be changed relative to the rest of the sentence. 

Even with a placeholder, this sort of string can still be problematic where the translation may vary depending on e.g. the gender of the noun of the placeholder value. In this case, I don't believe that should be a problem, given that the values to be substituted are proper names of companies / services.

That said, it would be good to:

- Confirm whether this is a correct assumption.
- Update to use a placeholder.
- Consider including a [translator comment](https://codex.wordpress.org/I18n_for_WordPress_Developers#Descriptions) to explain this.

Reply:
Issue for tracking purposes: #20152